### PR TITLE
[FxImporter] Add InputInfo to hooks

### PR DESCRIPTION
**Pull request by chrsmcgrr**
iree-turbine uses the FxImporter hooks to materialize buffers as global constants. In this particular use case we have a buffer that is modified and so requires the mutable modifier.

However, this information is not available in the current API as the InputInfo structure is not passed to the literal resolution. 

The following change passes the information to the literal API of the FxImporterHook.

### CODE REVIEW

**Comment by chrsmcgrr** [comment]
* **#1** <a href="https://app.graphite.dev/github/pr/RooflineAI/torch-mlir/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @chrsmcgrr and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python/torch_mlir/extras/fx_importer.py [hunk 1]
```diff
@@ -470,7 +470,7 @@ def prepare_module(self, module_op: Operation):
         ...
 
     def resolve_literal(
-        self, gni: "GraphNodeImporter", literal: Any
+        self, gni: "GraphNodeImporter", literal: Any, info: InputInfo
```
Comment (maxbartel) [inline python/torch_mlir/extras/fx_importer.py:473]: This needs to be optional, too

**Review by maxbartel — APPROVED** [review]
LGTM

File: python/torch_mlir/extras/fx_importer.py (hunk 1 above)
Comment (chrsmcgrr) [reply python/torch_mlir/extras/fx_importer.py]: done

**Pull request closed without merging**